src/OpenTelemetry/README.md

```diff
@@ -84,6 +84,51 @@ using var otel = Sdk.CreateTracerProvider(b => b
   * [Building your own Processor](../../docs/trace/extending-the-sdk/README.md#processor)
   * [Building your own Sampler](../../docs/trace/extending-the-sdk/README.md#sampler)
 
+### Troubleshooting
+
+Self diagnostics module is provided to help troubleshooting.
```

Comment:
Maybe add a bit more of an explanation here?

> Self diagnostics module is provided to help troubleshooting. When enabled internal events generated by OpenTelemetry will be written to a log file. These events contain important debugging information and error messages.

Reply:
Would be good to clarify two things:
1. Self-diagnostic modules subscribe to EventSource events with the name "OpenTelemetry.*"
2. All OpenTelemetry packages from this repo log to EventSource with a name of the pattern "OpenTelemetry.componentname". Hence enabling this will gather logs from all the components from this repo, and anything else emitting to EventSource "OpenTelemetry.*"

Reply:
Added.